### PR TITLE
test: add integration tests and input validation for auth routes

### LINKED ISSUE #67: [TEST] Unit Testing: authentication endpoint
# Test Task: `authentication endpoint`

## Goal
Add tests for `authentication endpoint` to ensure correct behavior and edge case handling.

---

## Test Type
- [x] Unit (logic / handler)
- [ ] E2E (full request flow)

---

## Done When
- All tests pass in CI or locally if pipeline testing not setup yet
- Core flows + edge cases covered
- No failing lint/tests

### PR DESCRIPTION
**Pull request by Fresh-Bruh**
CLOSES #67

### CODE REVIEW

File: issue-tracker-api/test/auth.spec.js [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+
+				// token from response
+				const { token } = await res.json();
+
+				// find session that matches the token and check it exists
+				const session = await env.DB.prepare('SELECT * FROM sessions WHERE token = ?').bind(token).first();
+				expect(session).not.toBeNull();
+			});
+
+			it('inserts a session row with the correct token, user_id, and a future expires_at', async () => {
```
Comment (Amormio25) [inline issue-tracker-api/test/auth.spec.js:113]: ima comment on these just to lyk, but if the tests pass it's fine.

This test is a bit redundant, the test just above already checked for existence of a session upon creation, now you're checking that the fields are correct. In practice you could just put them into the same test something like this tho this incomplete:

```bash
it('creates a valid session in the database', async () => {
    const res = await registerUser();
    expect(res.status).toBe(201);
    
    const { token, expires_at } = await res.json();
    const session = await env.DB.prepare('SELECT * FROM sessions WHERE token = ?').bind(token).first();
    
    expect(session).toBeDefined();
    expect(session.token).toBe(token);
    expect(session.user_id).toBeDefined();
    expect(session.expires_at).toBe(expires_at);
});
```
saves you time and lines of code

File: issue-tracker-api/test/auth.spec.js [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+		});
+	});
+
+	// ==========================================
+	// POST /auth/login Testing
+	// ==========================================
+	describe('POST /auth/login', () => {
+		describe('Success Cases', () => {
+			// Tests return values
+			it('returns a non-empty token and expires_at on valid credentials', async () => {
```
Comment (Amormio25) [inline issue-tracker-api/test/auth.spec.js:320]: same idea here some of these success tests can be grouped

**Review by Amormio25 — APPROVED** [review]
I assume you verified all test cases passed, looks good I just made a few comments about redundancy but you don't need to make those changes just an fyi

File: issue-tracker-api/test/auth.spec.js (hunk 1 above)
Comment (FreshAnemone) [reply issue-tracker-api/test/auth.spec.js:113]: I'll fix that.

File: issue-tracker-api/test/auth.spec.js (hunk 2 above)
Comment (FreshAnemone) [reply issue-tracker-api/test/auth.spec.js:320]: Resolved
